x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/PlannerSettings.java

```diff
@@ -45,26 +45,42 @@ public class PhysicalSettings {
         Setting.Property.Dynamic
     );
 
+    public static final Setting<ByteSizeValue> INTERMEDIATE_LOCAL_RELATION_MAX_SIZE = Setting.memorySizeSetting(
+        "esql.intermediate_local_relation_max_size",
+        "0.1%",
```

Comment:
On a 512mb heap this is half a megabyte. That feels fine.

Reply:
Yeh.. I dropped the min/max bounds for the limit, since I think this proportionality should be working by itself. But can add them back if it proves it'll be useful.